components/event-publisher-proxy/pkg/legacy-events/legacy.go

```diff
@@ -0,0 +1,206 @@
+package legacy
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	cev2event "github.com/cloudevents/sdk-go/v2/event"
+	"github.com/google/uuid"
+	apiv1 "github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/legacy-events/api"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	isValidEventTypeVersion = regexp.MustCompile(AllowedEventTypeVersionChars).MatchString
+	isValidEventID          = regexp.MustCompile(AllowedEventIDChars).MatchString
+)
+
+type Transformer struct {
+	bebNamespace    string
+	eventTypePrefix string
+}
+
+func NewTransformer(bebNamespace string, eventTypePrefix string) *Transformer {
+	return &Transformer{
+		bebNamespace:    bebNamespace,
+		eventTypePrefix: eventTypePrefix,
+	}
+}
+
+const (
+	requestBodyTooLargeErrorMessage = "http: request body too large"
+	eventTypeVersionExtensionKey    = "eventtypeversion"
+)
+
+func (t Transformer) TransformsLegacyRequestsToCE(writer http.ResponseWriter, request *http.Request) *cev2event.Event {
+
+	// parse request body to PublishRequestV1
+	if request.Body == nil || request.ContentLength == 0 {
+		resp := ErrorResponseBadRequest(ErrorMessageBadPayload)
+		writeJSONResponse(writer, resp)
+		return nil
+	}
+	appName := parseApplicationNameFromPath(request.URL.Path)
+	var err error
+	parameters := &apiv1.PublishEventParametersV1{}
+	decoder := json.NewDecoder(request.Body)
+	err = decoder.Decode(&parameters.PublishrequestV1)
+	if err != nil {
+		var resp *apiv1.PublishEventResponses
+		if err.Error() == requestBodyTooLargeErrorMessage {
+			resp = ErrorResponseRequestBodyTooLarge(err.Error())
+		} else {
+			resp = ErrorResponseBadRequest(err.Error())
+		}
+		writeJSONResponse(writer, resp)
+		return nil
+	}
+
+	// validate the PublishRequestV1 for missing / incoherent values
+	checkResp := checkParameters(parameters)
+	if checkResp.Error != nil {
+		writeJSONResponse(writer, checkResp)
+		return nil
+	}
+
+	response := &apiv1.PublishEventResponses{}
+
+	event, err := t.convertPublishRequestToCloudEvent(appName, parameters)
+	if err != nil {
+		response.Error = &apiv1.Error{
+			Status:  http.StatusInternalServerError,
+			Message: err.Error(),
+		}
+		writeJSONResponse(writer, response)
+		return nil
+	}
+
+	return event
+}
+
+func parseApplicationNameFromPath(path string) string {
+	// Assumption: Clients(application validator which has a flag for the path(https://github.com/kyma-project/kyma/blob/master/components/application-connectivity-validator/cmd/applicationconnectivityvalidator/applicationconnectivityvalidator.go#L49) using this endpoint must be sending request to path /:application/v1/events
+	// Hence it should be safe to return 0th index as the application name
+	pathSegments := make([]string, 0)
+	for _, segment := range strings.Split(path, "/") {
+		if strings.Trim(segment, " ") != "" {
```

Comment:
I recommend using `TrimSpace` to trim all whitespaces (e.g. `\t`...).

Reply:
Changed